File: SchoolWebApp-Client/src/app/finance/models/account.ts
import {ResourceModel} from '@/core/models/ResourceModel';

export enum AccountType {
    Asset = 1,
    Liability = 2,
    Equity = 3,
    Income = 4,
    Expense = 5
}

export class Account extends ResourceModel<Account> {
    public code?: string;
    public name?: string;
    public accountType?: AccountType;
    public parentAccountId?: number;
    public parentAccountName?: string;
    public description?: string;
    public isActive?: boolean;

    constructor(model?: Partial<Account>) {
        super(model);
    }
}

File: SchoolWebApp-API/Project.Core/Entities/Finance/Account.cs
using SchoolWebApp.Core.Entities.Shared;
using System.ComponentModel.DataAnnotations;

namespace SchoolWebApp.Core.Entities.Finance
{
    public class Account : Base
    {
        [Required]
        [StringLength(20)]
        public required string Code { get; set; }

        [Required]
        [StringLength(255)]
        public required string Name { get; set; }

        public AccountType AccountType { get; set; }

        public int? ParentAccountId { get; set; }
        public Account? ParentAccount { get; set; }

        [StringLength(500)]
        public string? Description { get; set; }

        public bool IsActive { get; set; } = true;

        public List<JournalLine> JournalLines { get; set; } = new();
    }
}
